### PR TITLE
test: Have GitHub workflows build functional tests

### PR DESCRIPTION
**Pull request by rajachan**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rauteric — APPROVED** [review]

**Comment by rajachan** [comment]
Failing CI actions are an artifact of how the distcheck build is setup, and it needs an update.

**Review by bwbarrett — APPROVED** [review]
Your commit subject line is too long, and it would be great to have a bit more in the commit message about the issue you're fixing.

**Comment by aws-nslick** [comment]
> Failing CI actions are an artifact of how the distcheck build is setup, and it needs an update. Will fix that up.

Please check the refactor of this I have in https://github.com/aws/aws-ofi-nccl/pull/420 and consider whether you can rebase atop it.

**Comment by rajachan** [comment]
Fixed CI and commit message.

**Review by rauteric — DISMISSED** [review]

**Comment by AvivBenchorin** [comment]
bot:aws:retest

File: configure.ac [hunk 1]
```diff
@@ -114,7 +114,7 @@ AC_ARG_ENABLE([tests],
 AS_IF([test "${enable_tests}" != "no" -a "${enable_neuron}" = "yes"],
       [AC_MSG_WARN([Disabling tests due to Neuron configuration.])
        enable_tests=no],
-      [test "${enable_tests}" = "yes" -a "${found_mpi}" = "no"],
+      [test "${enable_tests}" != "no" -a "${found_mpi}" = "no"],
```
Comment (bwbarrett) [inline configure.ac:117]: I don't think this is right, and it certainly isn't a fix (change yes, fix no).  We don't want to require MPI for normal users to build, and there's no real value in the tests for end users, so why force everyone to have MPI?

**Review by bwbarrett — DISMISSED** [review]

File: configure.ac (hunk 1 above)
Comment (rauteric) [reply configure.ac:117]: I'm not following. This line makes sure that if tests are enabled, MPI is found. There was a bug, in that `test "${enable_tests}" = "yes"` is not the correct way to check if tests are enabled.

Whether tests are enabled by default is a different question.

Comment (bwbarrett) [reply configure.ac:117]: `test "${enable_tests}" = "yes"` absolutely is the right way to check if the customer explicitly enabled the tests.  Why do you think it isn't?

Comment (rauteric) [reply configure.ac:117]: Ok, sorry, I understand what you're saying now.

> `test "${enable_tests}" = "yes"` absolutely is the right way to check if the customer explicitly enabled the tests.

Agree. The problem is that we are currently enabling tests by default, without the customer explicitly enabling them.

If the customer does not explicitly enable tests, `test "${enable_tests}" = "yes"` will evaluate to false, so we won't check for MPI. But we will try to build the tests anyway and fail during build-time.

So, we can either make this change, or change to tests-disabled-by-default. But the code today will pass configure (implicitly enabling tests) and then fail during build if MPI isn't found, which is wrong.

**Review by rauteric — APPROVED** [review]

**Review by aws-nslick — APPROVED** [review]

**Pull request merged**